pylxd/models/image.py

```diff
@@ -119,7 +122,17 @@ def create(
         if public:
             headers['X-LXD-Public'] = '1'
 
-        if metadata is not None:
+        if from_streams:
```

Comment:
Is there a compelling reason that we'd ever want to _not_ stream the image? If not, I'd prefer that we not add a new option and just chunk all image uploads

Reply:
I wrote it in comment - the only reason if we want to have pylxd clients to work with old version too.
For example nova-lxd - if it will be switched to stream based image upload it will require NEW version of pyxld i.e it will not be able to wold with old version of pylxd. I do not see any versioning in pylxd code so client will not be able which version of pyxld is installed.
As I was not sure that pyxld and nova-lxd will be updated synchronously by packagers I assume that clients can fallback to non-stream behavior via introspection of create method.
In short it will allow client of pyxld library to work with both version of pyxld - fixed one and old one. If we are sure that pyxld will be ALWAYS updated as it's client we can remove this for sure and assume that streams always passed to create operation.

Reply:
May be it make sense at least to dynamically identify if streams are passed or content as before and act accordingly?
It will provide backward compatibility for 'old' clients which will pass content instead of streams. What do you think?